src/components/scheduling/goals/SchedulingGoal.svelte

```diff
@@ -102,10 +119,36 @@
     }
   }
 
+  function getCleansedArguments(specArguments: Argument, schema?: ValueSchema) {
```

Comment:
Is the idea here that there could be old/outdated arguments in the db that could no longer be compatible/relevant to the current constraint?

Reply:
Yeah that's pretty close.  It's specifically for when the spec has arguments that aren't present in the new version of the goal/constraint that they switched to.

Reply:
Got it, maybe a small comment for future us/people on that function then?